data_generator.py

```diff
@@ -33,44 +35,37 @@ def on_epoch_end(self):
   def __len__(self):
     return int(len(self.df) / self.batch_size)
 
-  # scale RGB channels to [0,1]
-  def scale_input(self,img):
-    return img/255.0 
-
-  # resize img to input shape (will likely replace once we start dealing with bboxes)
-  def resize_img(self,img):
-    return cv2.resize(img,self.output_size,interpolation=cv2.INTER_LINEAR) # what interpolation is best?
-
-  # adjust x/y coords to new resized img
-  def resize_label(self,label,og_width,og_height):
+  def transform_image(self, img):
+    og_width, og_height = int(img.shape[1]), int(img.shape[0])
+    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB) # fix RBG so images are easier to plot
+    img = img/255.0 # scale RGB channels to [0,1]
+    img = cv2.resize(img,self.output_size,interpolation=cv2.INTER_LINEAR) # resize
+    return img, og_width, og_height
+  
+  def transform_label(self,label, og_width, og_height):
+    label = label.strip('][').split(', ')
     x_scale = self.output_size[0]/og_width
     y_scale = self.output_size[1]/og_height
-    return [int(round(int(val)*(x_scale if i%3==0 else y_scale if i%3==1 else 1))) for i,val in enumerate(label)]
+    # adjust x/y coords to new resized img
+    transformed_label = [int(round(int(val)*(x_scale if i%NUM_COCO_KP_ATTRBS==0 else y_scale if i%NUM_COCO_KP_ATTRBS==1 else 1))) for i,val in enumerate(label)]
+    return np.asarray(transformed_label)
 
   # returns batch at index idx
   def __getitem__(self, idx):
-    ## Initialize Batch:
+    # Initialize Batch:
     X = np.empty((self.batch_size, *self.output_size, 3)) # 3 channels of RGB
-    y = np.empty((self.batch_size, 3 * 17)) # (x,y,v) * 17 keypoints
+    y = np.empty((self.batch_size, NUM_COCO_KP_ATTRBS * NUM_COCO_KEYPOINTS))
```

Comment:
```suggestion
    y = np.empty((self.batch_size, *OUTPUT_DIM, NUM_COCO_KP_ATTRBS * NUM_COCO_KEYPOINTS))
```

Reply:
keep in mind for each frame we have a heat map of size `OUTPUT_DIM`

Reply:
This change should fold in when heatmap generation gets added, otherwise, pipeline breaks b/c label shape doesn't match yet.

Reply:
Change of shape added, resolving conversation